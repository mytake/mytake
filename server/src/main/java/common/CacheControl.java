/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import org.jooby.Response;

public class CacheControl {
	public static Response forever(Response res) {
		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
		return res.header("Cache-Control", "public, max-age=31536000, immutable");
	}

	public static Response hour(Response res) {
		// https://support.cloudflare.com/hc/en-us/articles/202775670-Customizing-Cloudflare-s-cache#:~:text=Cache%20additional%20content%20at%20Cloudflare,-Caching%20additional%20content&text=Do%20not%20use%20Cache%20Everything,Cache%20Level%20set%20to%20Bypass.
		return res.header("Cache-Control", "public, max-age=3600, immutable");
	}

	public static Response bypass(Response res) {
		// https://support.cloudflare.com/hc/en-us/articles/202775670-Customizing-Cloudflare-s-cache#:~:text=Cache%20additional%20content%20at%20Cloudflare,-Caching%20additional%20content&text=Do%20not%20use%20Cache%20Everything,Cache%20Level%20set%20to%20Bypass.
		return res.header("Cache-Control", "no-cache");
	}

	public static Response corsAllowAll(Response res) {
		return res.header("Access-Control-Allow-Origin", "*");
	}
}
