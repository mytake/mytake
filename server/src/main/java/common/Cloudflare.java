/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import org.jooby.Result;
import org.jooby.Results;

/** Sends a response with cache-headers which guarantee forever-immutability. */
public class Cloudflare {
	public static Result json(Object toSend) {
		return Results.json(toSend)
				.header("Cache-Control",
						"max-age=31536000", // one year https://stackoverflow.com/a/25201898/1153071
						"public", // any cache may store the response
						"no-transform", // don't muck with it at all
						"immutable" // it will never change at all for sure
				);
	}
}
