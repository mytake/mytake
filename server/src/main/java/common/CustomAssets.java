/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.google.common.io.Resources;
import com.google.inject.Binder;
import com.jsoniter.JsonIterator;
import com.jsoniter.any.Any;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.function.BiFunction;
import java.util.stream.Collectors;
import org.jooby.Env;
import org.jooby.Jooby;
import org.jooby.assets.AssetCompiler;
import org.jooby.rocker.Rockerby;

public class CustomAssets implements Jooby.Module {
	public static void initTemplates(Jooby jooby) {
		// sets variables that the templates need
		jooby.use(new CustomAssets());
		// makes rocker templates work
		jooby.use(new Rockerby());
	}

	private CustomAssets() {}

	static final String _STYLES = "_styles";
	static final String _SCRIPTS = "_scripts";
	static final String FB_APP_ID = "fbappid";

	@Override
	public void configure(Env env, Config conf, Binder binder) throws Throwable {
		env.router().assets("/favicon.ico", "/assets/permanent/favicon-6022c3e42d.ico");

		Config config;
		BiFunction<String, String, String> url;
		String fbAppId = conf.getString("application.fbAppId");
		if (env.name().equals("dev")) {
			config = ConfigFactory.parseResources("assets.dev.conf");
			url = (type, raw) -> "/assets-dev/" + type + raw;
			env.router().assets("/assets-dev/**");
			env.router().assets("/assets/**");
		} else {
			config = ConfigFactory.parseResources("assets.conf");
			byte[] manifest = Resources.toByteArray(CustomAssets.class.getResource("/assets/manifest.json"));
			Map<String, Any> map = JsonIterator.deserialize(manifest).asMap();
			url = (type, raw) -> "/assets/" + type + "/" + Objects.requireNonNull(map.get(raw.substring(1)), "No fingerprinted version of " + raw + ", only has: " + map.keySet());
			env.router().assets("/assets/**");
		}
		AssetCompiler compiler = new AssetCompiler(config);
		// key, style, key, script
		List<String> keyValue = keyValueFor(compiler, url);
		env.router().before((req, rsp) -> {
			for (int i = 0; i < keyValue.size() / 2; ++i) {
				String key = keyValue.get(2 * i);
				String value = keyValue.get(2 * i + 1);
				req.set(key, value);
			}
			req.set(FB_APP_ID, fbAppId);
		});
	}

	static List<String> keyValueFor(AssetCompiler compiler, BiFunction<String, String, String> url) {
		List<String> keyValue = new ArrayList<>(4 * compiler.fileset().size());
		for (String fileset : compiler.fileset()) {
			keyValue.add(fileset + CustomAssets._STYLES);
			keyValue.add(styles(url, compiler, fileset));
			keyValue.add(fileset + CustomAssets._SCRIPTS);
			keyValue.add(scripts(url, compiler, fileset));
		}
		return keyValue;
	}

	private static final String SLASH_SHA_384 = "/sha384-";

	private static String styles(BiFunction<String, String, String> urlMapper, AssetCompiler compiler, String fileset) {
		return compiler.styles(fileset).stream()
				.map(url -> url.startsWith(SLASH_SHA_384) ? url : urlMapper.apply("styles", url))
				.map(CustomAssets::subresourceIntegrityStyle)
				.collect(Collectors.joining("\n"));
	}

	private static String scripts(BiFunction<String, String, String> urlMapper, AssetCompiler compiler, String fileset) {
		return compiler.scripts(fileset).stream()
				.map(url -> url.startsWith(SLASH_SHA_384) ? url : urlMapper.apply("scripts", url))
				.map(CustomAssets::subresourceIntegrityScript)
				.collect(Collectors.joining("\n"));
	}

	private static String subresourceIntegrityStyle(String input) {
		if (input.startsWith(SLASH_SHA_384)) {
			int barIdx = input.indexOf('|');
			String integrity = input.substring(1, barIdx); // 1=remove slash
			String url = input.substring(barIdx + 1);
			return "<link rel=\"stylesheet\" href=\"" + url + "\" integrity=\"" + integrity + "\" crossorigin=\"anonymous\">";
		} else {
			return "<link rel=\"stylesheet\" href=\"" + input + "\">";
		}
	}

	private static String subresourceIntegrityScript(String input) {
		if (input.startsWith(SLASH_SHA_384)) {
			int barIdx = input.indexOf('|');
			String integrity = input.substring(1, barIdx); // 1=remove slash
			String url = input.substring(barIdx + 1);
			return "<script type=\"text/javascript\" src=\"" + url + "\" integrity=\"" + integrity + "\" crossorigin=\"anonymous\"></script>";
		} else {
			return "<script type=\"text/javascript\" src=\"" + input + "\"></script>";
		}
	}
}
