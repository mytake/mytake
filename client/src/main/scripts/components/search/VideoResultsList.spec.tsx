/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as renderer from "react-test-renderer";
import { FoundationHarness } from "../../common/foundationTest";
import { SearchMode, _searchImpl, _SearchWithData } from "./search";
// this data is generated by server/src/test/java/controllers/LucenePlayground.java
import socialSecuritySearchResults from "./testData/socialSecuritySearchResults.json";
import { VideoResultsList } from "./VideoResultsList";

test("VideoResultsList social security", () => {
  const result = _searchImpl(
    new _SearchWithData(
      "social security",
      socialSecuritySearchResults.facts,
      FoundationHarness.loadAllFromDisk(),
      SearchMode.BeforeAndAfter
    )
  );

  const tree = renderer
    .create(
      new VideoResultsList({
        mode: SearchMode.BeforeAndAfter,
        onModeChange: jest.fn(),
        searchResult: result,
      }).render()
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("VideoResultsList no results", () => {
  const result = _searchImpl(
    new _SearchWithData(
      "gibberish",
      [],
      FoundationHarness.loadAllFromDisk(),
      SearchMode.BeforeAndAfter
    )
  );
  const tree = renderer
    .create(
      new VideoResultsList({
        mode: SearchMode.BeforeAndAfter,
        onModeChange: jest.fn(),
        searchResult: result,
      }).render()
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
