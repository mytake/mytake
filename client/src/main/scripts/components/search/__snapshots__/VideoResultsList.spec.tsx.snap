// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoResultsList no results 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <h1
      className="results__heading"
    >
      0
       Search Results
    </h1>
    <p
      className="turn__results"
    >
      Search returned no results for 
      <strong>
        gibberish
      </strong>
    </p>
  </div>
</div>
`;

exports[`VideoResultsList social security 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <h1
      className="results__heading"
    >
      252
       Search Results
    </h1>
    <div>
      <div
        className=""
      >
        <div>
          NGramViewer: 
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <SearchRadioButtons
        onChange={[MockFunction]}
        selectedOption={1}
      />
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Kennedy, Nixon (1 of 4)
         - 
        1960-09-26
      </h2>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        38
         
        332
        552
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        38
         
        553
        1072
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        39
         
        358
        749
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        49
         
        1121
        1456
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        56
         
        1021
        1240
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        56
         
        1281
        1520
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        70
         
        1937
        2316
      </div>
      <div>
        VideoResult: 
        CsurHThBBkk
         
        72
         
        543
        851
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Kennedy, Nixon (3 of 4)
         - 
        1960-10-13
      </h2>
      <div>
        VideoResult: 
        PdjgZ5xx42o
         
        46
         
        1462
        2260
      </div>
      <div>
        VideoResult: 
        PdjgZ5xx42o
         
        48
         
        300
        666
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Carter, Ford (1 of 3)
         - 
        1976-09-23
      </h2>
      <div>
        VideoResult: 
        Zbe9FnAmkSk
         
        17
         
        0
        286
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Anderson, Reagan (1 of 2)
         - 
        1980-09-21
      </h2>
      <div>
        VideoResult: 
        pZqindVjd54
         
        12
         
        886
        1153
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Carter, Reagan (2 of 2)
         - 
        1980-10-28
      </h2>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        98
         
        0
        413
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        99
         
        0
        1139
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        101
         
        0
        408
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        102
         
        0
        1176
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        104
         
        28
        1101
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        106
         
        0
        900
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        108
         
        364
        524
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        108
         
        525
        958
      </div>
      <div>
        VideoResult: 
        a1J1TkWCLhs
         
        110
         
        0
        369
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Mondale, Reagan (1 of 2)
         - 
        1984-10-07
      </h2>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        9
         
        0
        338
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        11
         
        405
        942
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        13
         
        0
        509
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        13
         
        510
        692
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        24
         
        84
        350
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        24
         
        351
        762
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        26
         
        0
        521
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        26
         
        550
        805
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        26
         
        806
        1113
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        50
         
        146
        435
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        78
         
        781
        1021
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        78
         
        1138
        1644
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        88
         
        0
        388
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        110
         
        703
        997
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        112
         
        69
        704
      </div>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        116
         
        6
        716
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Dukakis (1 of 2)
         - 
        1988-09-25
      </h2>
      <div>
        VideoResult: 
        HXuffF9Sib0
         
        16
         
        0
        202
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Dukakis (2 of 2)
         - 
        1988-10-13
      </h2>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        11
         
        23
        353
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        21
         
        281
        621
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        97
         
        0
        442
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        100
         
        0
        121
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        100
         
        271
        834
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        100
         
        849
        1349
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        100
         
        1350
        1806
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        102
         
        0
        208
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        102
         
        721
        972
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        111
         
        0
        380
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        112
         
        0
        302
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        112
         
        852
        1563
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        114
         
        0
        1199
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        116
         
        0
        1123
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        120
         
        275
        560
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        122
         
        0
        386
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        125
         
        0
        194
      </div>
      <div>
        VideoResult: 
        KxvfacuEbnA
         
        126
         
        0
        65
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Clinton, Perot (1 of 3)
         - 
        1992-10-11
      </h2>
      <div>
        VideoResult: 
        9Z43XV4T8Bc
         
        46
         
        256
        509
      </div>
      <div>
        VideoResult: 
        9Z43XV4T8Bc
         
        124
         
        0
        545
      </div>
      <div>
        VideoResult: 
        9Z43XV4T8Bc
         
        124
         
        981
        1112
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Clinton, Perot (2 of 3)
         - 
        1992-10-15
      </h2>
      <div>
        VideoResult: 
        0Z9DZ715Ku4
         
        13
         
        1462
        1862
      </div>
      <div>
        VideoResult: 
        0Z9DZ715Ku4
         
        137
         
        33
        311
      </div>
      <div>
        VideoResult: 
        0Z9DZ715Ku4
         
        139
         
        0
        304
      </div>
      <div>
        VideoResult: 
        0Z9DZ715Ku4
         
        139
         
        733
        941
      </div>
      <div>
        VideoResult: 
        0Z9DZ715Ku4
         
        153
         
        926
        1359
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Clinton, Dole (1 of 2)
         - 
        1996-10-06
      </h2>
      <div>
        VideoResult: 
        JcyqYFlh_x4
         
        7
         
        452
        645
      </div>
      <div>
        VideoResult: 
        JcyqYFlh_x4
         
        21
         
        223
        434
      </div>
      <div>
        VideoResult: 
        JcyqYFlh_x4
         
        21
         
        786
        1099
      </div>
      <div>
        VideoResult: 
        JcyqYFlh_x4
         
        23
         
        0
        354
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Clinton, Dole (2 of 2)
         - 
        1996-10-16
      </h2>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        40
         
        42
        404
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        41
         
        0
        930
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        41
         
        1355
        1948
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        42
         
        118
        313
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        42
         
        410
        632
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        42
         
        977
        1179
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        55
         
        1038
        1127
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        76
         
        571
        749
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        106
         
        786
        1065
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        111
         
        0
        296
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        112
         
        111
        538
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        112
         
        574
        935
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        112
         
        953
        1053
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        113
         
        57
        488
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        113
         
        1244
        1433
      </div>
      <div>
        VideoResult: 
        T_ttLcdbAPE
         
        116
         
        251
        441
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Gore (1 of 3)
         - 
        2000-10-03
      </h2>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        3
         
        137
        407
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        9
         
        224
        518
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        14
         
        1382
        2221
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        95
         
        957
        1262
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        95
         
        1331
        1561
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        130
         
        909
        1458
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        131
         
        495
        772
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        132
         
        755
        1045
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        134
         
        0
        243
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        137
         
        178
        683
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        137
         
        747
        1526
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        137
         
        1527
        1711
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        138
         
        0
        490
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        138
         
        727
        1040
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        138
         
        1158
        1742
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        139
         
        0
        354
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        139
         
        355
        536
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        140
         
        0
        125
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        140
         
        169
        291
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        140
         
        595
        775
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        143
         
        0
        181
      </div>
      <div>
        VideoResult: 
        NQFabXU_52Q
         
        144
         
        225
        483
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Gore (3 of 3)
         - 
        2000-10-17
      </h2>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        85
         
        130
        300
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        140
         
        0
        239
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        181
         
        1022
        1357
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        183
         
        156
        326
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        183
         
        1081
        1405
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        185
         
        203
        483
      </div>
      <div>
        VideoResult: 
        ZFbyKIUp1E8
         
        185
         
        646
        1159
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Bush, Kerry (3 of 3)
         - 
        2004-10-13
      </h2>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        56
         
        4
        250
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        57
         
        0
        223
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        57
         
        832
        1357
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        59
         
        0
        552
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        59
         
        1227
        1326
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        59
         
        1356
        1441
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        60
         
        0
        411
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        61
         
        184
        632
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        61
         
        1162
        1361
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        61
         
        1853
        1995
      </div>
      <div>
        VideoResult: 
        lyj-qVXQMdI
         
        63
         
        0
        155
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - McCain, Obama (2 of 3)
         - 
        2008-10-07
      </h2>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        27
         
        0
        143
      </div>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        43
         
        0
        413
      </div>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        43
         
        414
        741
      </div>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        44
         
        95
        497
      </div>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        44
         
        1745
        2268
      </div>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        46
         
        6
        109
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Obama, Romney (1 of 3)
         - 
        2012-10-03
      </h2>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        62
         
        123
        206
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        63
         
        0
        234
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        63
         
        235
        498
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        63
         
        783
        1043
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        63
         
        2064
        2302
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        64
         
        0
        106
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        65
         
        0
        432
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        65
         
        659
        804
      </div>
      <div>
        VideoResult: 
        --oqXCLAbLg
         
        65
         
        1908
        2199
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Obama, Romney (2 of 3)
         - 
        2012-10-16
      </h2>
      <div>
        VideoResult: 
        nchGzjFHB-g
         
        120
         
        113
        249
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Clinton, Trump (3 of 3)
         - 
        2016-10-19
      </h2>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        318
         
        11
        818
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        322
         
        0
        313
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        323
         
        0
        357
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        325
         
        150
        565
      </div>
    </div>
  </div>
</div>
`;

exports[`VideoResultsList wall, -wall street 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <h1
      className="results__heading"
    >
      14
       Search Results
    </h1>
    <div>
      <div
        className=""
      >
        <div>
          NGramViewer: 
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <SearchRadioButtons
        onChange={[MockFunction]}
        selectedOption={1}
      />
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Mondale, Reagan (1 of 2)
         - 
        1984-10-07
      </h2>
      <div>
        VideoResult: 
        L7MdD72ZqJ0
         
        43
         
        87
        643
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - McCain, Obama (2 of 3)
         - 
        2008-10-07
      </h2>
      <div>
        VideoResult: 
        57qocCPgp3U
         
        38
         
        0
        200
      </div>
    </div>
    <div
      className="results__preview"
    >
      <h2
        className="results__subheading"
      >
        Presidential Debate - Clinton, Trump (3 of 3)
         - 
        2016-10-19
      </h2>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        31
         
        217
        354
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        32
         
        1534
        1694
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        34
         
        1437
        1853
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        38
         
        298
        515
      </div>
      <div>
        VideoResult: 
        eXNnggT5fPM
         
        45
         
        0
        13
      </div>
    </div>
  </div>
</div>
`;
